### PR TITLE
make matching a strict subset of labels

### PR DESCRIPTION
**Pull request by eiais**
Signed-off-by: Kyle Spiers <kyle@spiers.me>

### CODE REVIEW

**Comment by eiais** [comment]
closes #158

**Comment by kisom** [comment]
@eiais it looks like this caused the build to fail.

**Comment by eiais** [comment]
@kisom Yeah, there's a test in core_test that needs fixing. Also this could be a breaking change for some users I think. Is that something you would be concerned with merging?

File: keycache/keycache.go [hunk 1]
```diff
@@ -65,14 +65,20 @@ func (usage Usage) matchesLabel(labels []string) bool {
 		return true
 	}
 
-	for _, validLabel := range usage.Labels {
-		for _, label := range labels {
-			if label == validLabel {
-				return true
+	//Given an asset usage label must be a superset
```
Comment (kisom) [inline keycache/keycache.go]: Though this code predates some of our best practices (and the comment does match the style of other comments), it should follow standard Go practices now; e.g.

```
// Given an asset, usage label must be a superset.
```

**Comment by eiais** [comment]
@kisom Should be good to go now.

**Comment by kisom** [comment]
@eiais I'm going to have to do an audit of some our usage of RO this week to figure out if this will negatively impact anything, I'll keep you updated.

**Comment by eiais** [comment]
@kisom Any update?

**Comment by eiais** [comment]
Should I close this PR?

**Comment by eiais** [comment]
@grittygrease Can I get an update on this?

**Pull request closed without merging**